.github/workflows/ci.yml

```diff
@@ -8,35 +8,55 @@ on:
     branches:
       - "main"
       - "master"
+
 jobs:
-  test:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
+  run:
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
-        with:
-          python-version: "3.10"
+    runs-on: ${{ matrix.config.os }}
 
-      - name: Install dependencies
-        # https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-python#installing-dependencies
-        # If installing a development version of constellation, use:
-        # - pip3 install git+https://github.com/reside-ic/constellation@reside-62#egg=constellation
-        run: |
-          python -m pip install --upgrade setuptools pip wheel
-          pip3 install pytest-cov pycodestyle codecov
-          pip3 install -r requirements.txt
+    name: ${{matrix.config.os }} (${{ matrix.config.py }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: ubuntu-latest,  py: '3.8'}
+          - {os: ubuntu-latest,  py: '3.9'}
+          - {os: ubuntu-latest,  py: '3.10'}
+          - {os: ubuntu-latest,  py: '3.11'}
+          - {os: ubuntu-latest,  py: '3.12'}
 
-      - name: Tests
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.config.py }}
+        uses: actions/setup-python@v4
```

Comment:
These two actions (checkout, setup-python) have later versions available but perhaps those are not relevant